src/transformers/integrations/ggml.py

```diff
@@ -490,9 +516,9 @@ def __init__(self, dict_):
         for k, v in dict_.items():
             setattr(self, k, v)
 
-        if not hasattr(self, "tokens") or not hasattr(self, "scores"):
+        if (not hasattr(self, "tokens") or not hasattr(self, "scores")) and not hasattr(self, "merges"):
             raise ValueError("tokens and scores need to be passed for a LLaMa tokenizer to be instantiated.")
```

Comment:
can you clarify the error message here to state that either tokens and scores or merges needs to be passed for a GGUF tokenizer to be instantiated

Reply:
I think we can simplify this like:
```python3
if not hasattr(self, "merges"):
    if not hasattr(self, "tokens") or not hasattr(self, "scores"):
        raise ValueError("tokens and scores need to be passed for a LLaMa tokenizer without merges to be instantiated.")
    tokens = self.tokens
    scores = self.scores
    vocab = {t: scores[i] for i, t in enumerate(tokens)}

    logger.warning("Merges were not in checkpoint, building merges on the fly.")
    # building merges
    ...
```